### PR TITLE
add index photo widget to mobile version

### PR DESCRIPTION
**Pull request by spooner77**
add index photo widget to mobile version skalfa/workflow#107

### CODE REVIEW

File: mobile/components/index_photo_list.php [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+        $items = array('latest', 'toprated');
+        if ( $featured )
+        {
+            $items[] = 'featured';
+        }
+        
+        $menuItems = $this->getMenuItems($items, $uniqId);
+        $this->assign('items', $menuItems);
+
+        if ( !$latest && !OW::getUser()->isAuthorized('photo', 'upload') )
```
Comment (greyexpert) [inline mobile/components/index_photo_list.php]: Something wrong with this logic. Why are you checking `upload` permissions?

File: mobile/components/index_photo_list.php [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+
+        return $menuItems;
+    }
+    
+    public static function getToolbar($uniqId)
+    {
+        $lang = OW::getLanguage();
+        
+        $items = array('latest', 'toprated');
+        $url = OW::getEventManager()->call('photo.getAddPhotoURL');
```
Comment (greyexpert) [inline mobile/components/index_photo_list.php]: `$url` is unused in this scope

File: mobile/components/index_photo_list.php [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+        }
+
+        return $menuItems;
+    }
+    
+    public static function getToolbar($uniqId)
+    {
+        $lang = OW::getLanguage();
+        
+        $items = array('latest', 'toprated');
```
Comment (greyexpert) [inline mobile/components/index_photo_list.php]: `featured` missed in the array of toolbar items

File: mobile/components/photo_list_widget.php [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+{
+    /**
+     * @param BASE_CLASS_WidgetParameter $paramObj
+     */
+    public function __construct( BASE_CLASS_WidgetParameter $paramObj )
+    {
+        parent::__construct();
+
+        $photoService = PHOTO_BOL_PhotoService::getInstance();
+        $num = isset($paramObj->customParamList['photoCount']) ? $paramObj->customParamList['photoCount'] : 8;
```
Comment (greyexpert) [inline mobile/components/photo_list_widget.php]: Better use `empty()` - `isset()` will be always true

File: mobile/components/photo_list_widget.php [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+            'showTitle' => false,
+            'showToolbar' => true
+        );
+        
+        $uniqId = $cmpParams['uniqId'];
+        
+        $cmp = new PHOTO_MCMP_IndexPhotoList($cmpParams);
+        $this->addComponent('cmp', $cmp);
+        
+        $items = array('latest', 'toprated');
```
Comment (greyexpert) [inline mobile/components/photo_list_widget.php:63]: Add `featured` list

File: mobile/components/photo_list_widget.php [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+        $paramObj->standartParamList->capContent = (new BASE_MCMP_WidgetMenu($menuItems))->render();
+        
+        $this->setSettingValue(self::SETTING_TOOLBAR, PHOTO_MCMP_IndexPhotoList::getToolbar($uniqId));
+        $script = '';
+        
+        foreach($menuItems as $key => $item)
+        {
+            if ( !empty($item["visibility"]) )
+            {
+                $script .= '
```
Comment (greyexpert) [inline mobile/components/photo_list_widget.php:80]: Refactor this loop to not copy js snippets for each item

File: mobile/controllers/photo.php [hunk 7]
```diff
@@ -67,71 +67,22 @@ public function viewList()
...
-    }
-
-    public function viewToplist()
-    {
-        // is moderator
-        $modPermissions = OW::getUser()->isAuthorized('photo');
-
-        if ( !$modPermissions && !OW::getUser()->isAuthorized('photo', 'view') )
+        
+        $validLists = array('latest','toprated');
```
Comment (greyexpert) [inline mobile/controllers/photo.php]: `featured` is missing again

File: mobile/views/components/index_photo_list.html [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+{/literal}{/style}
+{capture assign='nocontent'}
+    <div class="owm_nocontent">{text key='photo+no_photo'}</div>
+{/capture}
+
+{capture assign='cmp'}
+    <div id="photo_list_cmp{$uniqId}">
+        <div class="owm_index_photo_list" id="{$items.latest.contId}">
+            {foreach from=$latest item='p'}     
+                <div class="owm_index_photo_item{if !empty($data.class)} {$data.class}{/if}">
```
Comment (greyexpert) [inline mobile/views/components/index_photo_list.html]: Could not find where you assigned `$data` to template. It seems that the variable is `null`

File: update/9113/update.php [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+ * INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+$widgetService = BOL_ComponentAdminService::getInstance();
```
Comment (greyexpert) [inline update/9113/update.php]: You should use `Updater::getWidgetService` here

File: mobile/components/photo_list_widget.php (hunk 6 above)
Comment (spooner77) [inline mobile/components/photo_list_widget.php:80]: There is maximum 3 items in the cycle

**Pull request merged**